rfc/0054-sdk3-management-apis.md

```diff
@@ -3070,6 +3072,46 @@ Nothing
 
     POST http://localhost:8091/pools/default/buckets/<bucket>/collections/<scope_name> -d name=<collection_name> -d maxTTL=<maxExpiry>
 
+## UpdateCollection
+
+Creates a new collection.
+
+### Signature
+
+```
+void UpdateCollection(CollectionSpec collection, [options])
+```
+
+### Parameters
+
+* Required:
+
+  * `collection`: [`CollectionSpec`](#collectionspec) - specification of the collection.
+
+* Optional:
+
+  * `Timeout` or `timeoutMillis` (`int`/`duration`) - the time allowed for the operation to be terminated. This is controlled by the client.
+
+  * `maxExpiry` (`duration`) - Value for the max expiry of new documents created without an expiry.
```

Comment:
Unsure what other SDKs have done but Go certainly doesn't.

Reply:
Absolute expiry does not make sense in the context of collection maxTTL.

For document CRUD operations we can allow either relative or absolute expiry, because the server ultimately converts a relative expiry to to an absolute expiry, and there's a simple and intuitive way to convert a relative expiry to an absolute expiry: just add the relative expiry to the current time, and viola!

For collection maxTTL, you can't say "I want all documents in the collection to expire on January 1st, 2024". That's not how the maxTTL field works. maxTTL is _always relative_ to the time when the document was created/modified.